src/main/java/com/adyen/model/terminal/SaleToAcquirerData.java

```diff
@@ -218,6 +221,8 @@ public String toString() {
 
     public String toBase64() {
         String json = PRETTY_PRINT_GSON.toJson(this);
-        return new String(Base64.encodeBase64(json.getBytes()));
+        Gson gson = new Gson();
+        String jsonOrdered = gson.toJson((Util.jsonObjectStringToTreeMap(json)), new TypeToken<HashMap>() { }.getType()); // Convert the JSONObject-like string to a treemap and cast it back to string to avoid element permutation
+        return new String(Base64.encodeBase64(jsonOrdered.getBytes()));
```

Comment:
This solution introduces some overhead by JSON encoding the data twice. 
```suggestion
        return new String(Base64.encodeBase64(json.getBytes()));
```